app/workers/work_packages/update_progress_job.rb

```diff
@@ -0,0 +1,257 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2024 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+# rubocop:disable Rails/SquishedSQLHeredocs
+class WorkPackages::UpdateProgressJob < ApplicationJob
+  queue_with_priority :default
+
+  def perform(current_mode:, previous_mode:)
+    with_temporary_progress_table do
+      if current_mode == "field"
+        unset_all_percent_complete_values if previous_mode == "disabled"
+        fix_remaining_work_set_with_100p_complete
+        fix_remaining_work_exceeding_work
+        fix_only_work_being_set
+        fix_only_remaining_work_being_set
+        derive_unset_remaining_work_from_work_and_p_complete
+        derive_unset_work_from_remaining_work_and_p_complete
+        derive_p_complete_from_work_and_remaining_work
+      end
+      if current_mode == "status"
```

Comment:
```suggestion
      elsif current_mode == "status"
```

I don't think we change `current_mode` in the first if block, so the second block can never be executed. This way we will not even need to look at the second if block.

Reply:
You're right. I used a case statement instead. Reads better IMO.
8c5c2094919